slider-core/src/main/java/org/apache/slider/providers/agent/AgentUtils.java

```diff
@@ -40,30 +41,54 @@
 public class AgentUtils {
   private static final Logger log = LoggerFactory.getLogger(AgentUtils.class);
 
+  public static Metainfo getApplicationMetainfoFromSummaryFile(
+      SliderFileSystem fileSystem,
+      String metainfoPath,
+      boolean metainfoForAddon
+      ) {
+    FileSystem fs = fileSystem.getFileSystem();
+    Path appPathXML = new Path(metainfoPath + ".metainfo.xml");
+    Path appPathJson = new Path(metainfoPath + ".metainfo.json");
+    try {
+      FSDataInputStream appStream = null;
+      if(fs.exists(appPathXML)) {
+        appStream = fs.open(appPathXML);
+        return parseMetainfo(appStream, metainfoForAddon, "xml");
+      }
+      else if(fs.exists(appPathJson)) {
+        appStream = fs.open(appPathJson);
+        return parseMetainfo(appStream, metainfoForAddon, "json");
+      }
+    } catch (IOException e) {
+      log.info("Failed to get metainfo from acompany file. {}", e.getMessage());
```

Comment:
1.s/acompany/r/accompanying file
2. include path.
3. add a log@debug with the stack trace